src/wp-includes/revision.php

```diff
@@ -516,6 +516,51 @@ function wp_get_post_revisions( $post_id = 0, $args = null ) {
 	return $revisions;
 }
 
+/**
+ * Returns last revisions of specified post and count of total revisions.
+ *
+ * @since 6.1.0
+ *
+ * @param int|WP_Post $post_id Optional. Post ID or WP_Post object.
+ * @return array {
+ *     Returns associative array with last revision and total count.
+ *
+ *     @type int $revision The last revision post id or 0 if non existing.
+ *     @type int $count The total count of revisions for $post_id.
+ * }
+ */
+function wp_get_last_revision_id_and_total_count( $post_id = 0 ) {
+	$post     = get_post( $post_id );
+	$revision = 0;
+	$count    = 0;
+
+	if ( ! $post || empty( $post->ID ) || ! wp_revisions_enabled( $post ) ) {
```

Comment:
```suggestion
	if ( ! $post || ! wp_revisions_enabled( $post ) ) {
```

I don't think this is needed. If post is truthy it should always have an ID.